examples/neighbors/plot_classification.py

```diff
@@ -48,11 +48,22 @@
     plt.contourf(xx, yy, Z, cmap=cmap_light)
 
     # Plot also the training points
-    plt.scatter(X[:, 0], X[:, 1], c=y, cmap=cmap_bold,
-                edgecolor='k', s=20)
+    scatter = plt.scatter(X[:, 0], X[:, 1], c=y, cmap=cmap_bold,
+                          edgecolor='k', s=20)
```

Comment:
You can just use seaborn here. It will take care of the legend. You need to import it on the top of the file.

```suggestion
    sns.scatterplot(x=X[:, 0], y=X[:, 1], hue=iris.target_names[y])
```